### PR TITLE
pos-print: printing receipts

### LINKED ISSUE #9: pos-print: endpoints for printing of receipts
Client Apps should be able to print receipts using the provided endpoints: 

Print Receipt

**POST /v1/posprint**
```json
{
  "sourceIp": "89.100.10.5",
  "operatorId": "??",
  "receipt": {"receiptId": "123", "currency": "BGN", "prefixLines": ["Customer: John"], "suffixLines": [], "amount": 20.00, "receiptItems": [{"name": "HSI 80/40", "quantity": 1.0, "price": 0.10}]}}
```

### PR DESCRIPTION
**Pull request by martinmilev**
Configured endpoint for printing receipts

Fixes #9

### CODE REVIEW

File: pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    })
+
+    val reply = service.printReceipt(request.mockRequest(recipitDto))
+    assertThat(reply,isOk)
+  }
+
+  @Test
+  fun printFiscalReceipt() {
+    val recipitDto = PrintService.ReceiptDTO("sourceIp", "::any ip::", true)
+    val receipt = Receipt.newReceipt().build()
```
Comment (mgenov) [inline pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt]: anyReceipt as it's irrelevant.

File: pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  private val factory = context.mock(PrinterFactory::class.java)
+  private val printer = context.mock(ReceiptPrinter::class.java)
+
+  private val request = SiteBricksRequestMockery()
+  private val service = PrintService(factory)
+
+  @Test
+  fun printReceipt() {
+    val recipitDto = PrintService.ReceiptDTO("sourceIp", "::any ip::", false)
+    val receipt = Receipt.newReceipt().build()
```
Comment (mgenov) [inline pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt]: anyReceipt

File: pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        oneOf(printer).close()
+      }
+    })
+
+    val reply = service.printReceipt(request.mockRequest(recipitDto))
+    assertThat(reply,isOk)
+  }
+
+  @Test
+  fun wrongSourceIp() {
```
Comment (mgenov) [inline pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt]: cashRegisterWasNotFound or unknownCashRegister

File: pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      }
+    })
+
+    val reply = service.printReceipt(request.mockRequest(recipitDto))
+    assertThat(reply, isBadRequest)
+    assertThat(reply, contains(ErrorResponse("Device not found.")))
+  }
+
+  @Test
+  fun cantConnect() {
```
Comment (mgenov) [inline pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt]: connectionCannotBeEstablished

File: pos-print/build.gradle [hunk 5]
```diff
@@ -61,7 +60,7 @@ dependencies {
   compile('com.google.sitebricks:sitebricks-client:0.8.11') {
     exclude module: 'guice'
     exclude module: 'guice-servlet'
-    exclude module: 'jackson-databind'
+//    exclude module: 'jackson-databind'
```
Comment (mgenov) [inline pos-print/build.gradle]: The commented line need to be removed if jackson is no longer going to be ignored.

File: pos-print/build.gradle [hunk 6]
```diff
@@ -72,7 +71,7 @@ dependencies {
     exclude module: 'guice'
     exclude module: 'guice-servlet'
     exclude module: 'guice-multibindings'
-    exclude module: 'jackson-databind'
+//    exclude module: 'jackson-databind'
```
Comment (mgenov) [inline pos-print/build.gradle]: The commented line need to be removed if jackson is no longer going to be ignored.

Comment (martinmilev) [inline pos-print/build.gradle]: removed

File: pos-print/build.gradle (hunk 5 above)
Comment (martinmilev) [inline pos-print/build.gradle]: removed

File: pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt (hunk 4 above)
Comment (martinmilev) [inline pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt]: renamed.

File: pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt (hunk 3 above)
Comment (martinmilev) [inline pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt]: renamed.

File: pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt (hunk 1 above)
Comment (martinmilev) [inline pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt]: renamed.

File: pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt (hunk 2 above)
Comment (martinmilev) [inline pos-print/src/test/java/com/clouway/pos/print/adapter/http/PrintServiceTest.kt]: renamed.

**Comment by martinmilev** [comment]
The PR is ready for review

File: pos-print/src/main/java/com/clouway/pos/print/core/ReceiptPrinter.java [hunk 7]
```diff
@@ -24,4 +24,11 @@
    * @throws IOException in case of failure
    */
   void printFiscalReceipt(Receipt receipt) throws IOException;
+
+  /**
+   * Closing output stream and socket
```
Comment (mgenov) [inline pos-print/src/main/java/com/clouway/pos/print/core/ReceiptPrinter.java]: Closes communication with the printer.

Comment (martinmilev) [inline pos-print/src/main/java/com/clouway/pos/print/core/ReceiptPrinter.java]: fixed

**Pull request merged**